### DISCUSSION TITLE
Clarity Datagrid (Hide/Show) Columns

### DISCUSSION THREAD

**Discussion by DrunkenApollo** [vmware-clarity/ng-clarity#1254] [Q&A] [answered]
Is there a way to track the state of the grids columns?
This is specifically for when selecting which columns to hide or show.
As I want to save this as preferences to be loaded when a user logs in etc...

**Comment by kevinbuhmann** [comment, marked as answer]
Yes, there is a `clrDgHiddenChange` event that you can handle. Here is an example: https://stackblitz.com/edit/clarity-light-theme-clr17-yua7jo

My example uses `[(clrDgHidden)]` for two-way binding, but you can de-sugar it to call an event handler method if needed: `[clrDgHidden]="value" (clrDgHiddenChange)="onHiddenChange(...)"`.
